### ISSUE TITLE
伪影分支和语义分支在每个数据集的具体数据

### ISSUE THREAD

**Issue by c401026366-design** [Megum1/CO-SPY#15]
作者您好，感谢您开源 CO-SPY 的代码与数据集。
我想请教一下，是否可以提供语义分支（semantic branch）和伪影分支（artifact branch）在 Co-Spy-Bench（包含5个真实数据集和22个测试数据集）上的具体评测结果（例如分数据集的 AP / Accuracy）？
另外，我在复现伪影分支时发现，其在 PG-v2-256 数据集上的表现较低，平均仅为 46.09 / 45.42（AP / Accuracy），接近随机水平。
我在 issue 中附上了我的实验结果以及论文/仓库中的对比结果。
从对比来看，我的结果与作者报告的结果存在明显差距，因此想请教：
* 这种在部分数据集上接近随机的现象是否是预期情况？
* 还是更可能与评测流程（如数据预处理、采样方式等）存在差异有关？
此外，我还想请问一下：是否有 Co-Spy-Bench/in-the-wild 设置下的完整评测流程或代码实现（例如数据预处理、JPEG设置、推理流程等）可以参考？
目前我主要是根据仓库已有代码自行整理实现，担心与作者原始设置存在细节差异。
非常感谢您的帮助。

<img width="1796" height="174" alt="Image" src="https://github.com/user-attachments/assets/44b1e09b-0f50-4efd-a90d-fb9d6aa7d795" />

<img width="470" height="420" alt="Image" src="https://github.com/user-attachments/assets/f733af18-6d96-4717-8724-40769459ee1f" />

**Comment by Youkeen** [comment]
你好，请问一下，你的结果是直接使用预训练权重还是自己训练的，我自己按照作者的训练方式，为什么ACC只有60多，但是使用预训练权重就是80多。

**Comment by Hil-create** [comment]
> 你好，请问一下，你的结果是直接使用预训练权重还是自己训练的，我自己按照作者的训练方式，为什么ACC只有60多，但是使用预训练权重就是80多。

我是用作者的权重训练的

**Comment by Megum1** [comment]
> 作者您好，感谢您开源 CO-SPY 的代码与数据集。 我想请教一下，是否可以提供语义分支（semantic branch）和伪影分支（artifact branch）在 Co-Spy-Bench（包含5个真实数据集和22个测试数据集）上的具体评测结果（例如分数据集的 AP / Accuracy）？ 另外，我在复现伪影分支时发现，其在 PG-v2-256 数据集上的表现较低，平均仅为 46.09 / 45.42（AP / Accuracy），接近随机水平。 我在 issue 中附上了我的实验结果以及论文/仓库中的对比结果。 从对比来看，我的结果与作者报告的结果存在明显差距，因此想请教：
> 
> * 这种在部分数据集上接近随机的现象是否是预期情况？
> * 还是更可能与评测流程（如数据预处理、采样方式等）存在差异有关？
>   此外，我还想请问一下：是否有 Co-Spy-Bench/in-the-wild 设置下的完整评测流程或代码实现（例如数据预处理、JPEG设置、推理流程等）可以参考？
>   目前我主要是根据仓库已有代码自行整理实现，担心与作者原始设置存在细节差异。
>   非常感谢您的帮助。
> 
> <img alt="Image" width="1796" height="174" src="https://private-user-images.githubusercontent.com/278442421/582086967-44b1e09b-0f50-4efd-a90d-fb9d6aa7d795.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzc4NTg3MTcsIm5iZiI6MTc3Nzg1ODQxNywicGF0aCI6Ii8yNzg0NDI0MjEvNTgyMDg2OTY3LTQ0YjFlMDliLTBmNTAtNGVmZC1hOTBkLWZiOWQ2YWE3ZDc5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUwNFQwMTMzMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMWE5OTc5OWZjYzM5MWM2OGEyZTNhZGJlYzg3YzExOTY1MTAwYWU2MDNmNmNjYzEzNzI4ZTg5MTRmMWNlNjRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.MF7vHi28Z32oKEuBSmndrRXJMdCPAcUfcWLgABAMbsQ"> <img alt="Image" width="470" height="420" src="https://private-user-images.githubusercontent.com/278442421/582089380-f733af18-6d96-4717-8724-40769459ee1f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzc4NTg3MTcsIm5iZiI6MTc3Nzg1ODQxNywicGF0aCI6Ii8yNzg0NDI0MjEvNTgyMDg5MzgwLWY3MzNhZjE4LTZkOTYtNDcxNy04NzI0LTQwNzY5NDU5ZWUxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUwNFQwMTMzMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMjJmNjI0MTRjMjY4MjZkMDM4OTA3NzdkZGJlMDYwODgyYmFiMWY3Y2M2MWNlYTk5ZjQ1MDY1N2NjM2JkOTJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.cvGs6mtF5FDmdLvg5qPvsaFIPbKPI0oqJKWYXlXTJuQ">

> 作者您好： 我正在复现您的 CO-SPY 论文工作，有一个地方想向您请教：论文中语义分支的特征插值与软标签样本构造的相关代码，我在 GitHub 仓库里仔细查找后没有找到，想请问这部分逻辑是在哪个文件 / 哪个函数里实现的呢？ 非常感谢您的工作与解答！

感谢您对 CO-SPY 的关注，也感谢您详细提供复现实验结果和对比信息。这些反馈对我们非常有帮助。

关于您提到的问题，我们这边做了以下说明和更新：

1. **关于单个分支的评测结果**

我们已经更新了 `main.py`，现在支持对单个分支进行单独测试。例如，可以使用如下命令测试 semantic branch 在 Co-Spy-Bench 上的结果：

```bash
python main.py --gpu 0 --phase eval --train_dataset sd-v1_4 --mode branch --branch semantic --pretrain
```

其中 `--branch` 可以设置为 `semantic` 或 `artifact`，分别用于测试语义分支和伪影分支。

同时，我们也提供了 `semantic_result.json` 和 `artifact_result.json`，其中包含两个分支在 Co-Spy-Bench 上的具体评测结果，供您参考和对比。

2. **关于 PG-v2-256 上复现结果较低的问题**

您观察到 artifact branch 在 PG-v2-256 上接近随机水平的现象，很可能与我们之前提供的 MSCOCO2017 训练集版本有关。

之前仓库中提供的 MSCOCO2017 训练集是从一个 HuggingFace repo 下载并处理的，该版本与原始数据在后处理细节上存在一些差异，这可能会影响复现结果。对此我们非常抱歉，也感谢您指出这一问题。

请您参考以下脚本重新更新训练集：

https://github.com/Megum1/CO-SPY/blob/main/data/train/sd-v1_4/download.sh

更新训练集后，artifact branch 的复现结果应该会更接近我们报告的结果。

3. **关于 Co-Spy-Bench / in-the-wild 的评测流程**

in-the-wild 设置下的数据处理流程与测试阶段基本一致。您可以参考测试数据集的实现：

https://github.com/Megum1/CO-SPY/blob/main/dataSets/dataset.py#L60

https://github.com/Megum1/CO-SPY/blob/main/detectors/sd-v1_4/fusion.py#L88

其中包含了测试数据读取以及相关后处理流程。JPEG 设置、推理流程等也可以按照测试流程中的实现进行整理。

希望以上说明和更新能够帮助到您！

[artifact_result.json](https://github.com/user-attachments/files/27327542/artifact_result.json)
[semantic_result.json](https://github.com/user-attachments/files/27327543/semantic_result.json)
